doc/manual/src/command-ref/files/profiles.md

```diff
@@ -21,11 +46,27 @@ lrwxrwxrwx 1 alice users 51 Oct 29 13:20 /home/alice/.local/state/nix/profiles/p
 lrwxrwxrwx 1 alice users 51 Nov 25 14:35 /home/alice/.local/state/nix/profiles/profile-7-link -> /nix/store/mp0x6xnsg0b8qhswy6riqvimai4gm677-profile
 ```
 
-Each of these symlinks is a root for the Nix garbage collector.
+- `~alice/.local/state/nix/profiles` is the directory that contains the profile.
+
+- `profile` is the name of the profile.
+
+- Generations 5, 6, and 7, currently exist.
+  (Lower numbered generations starting from 1 presumably once existed, but were deleted.)
 
-The contents of the store path corresponding to each version of the
-profile is a tree of symlinks to the files of the installed packages,
-e.g.
+- Generation 7 is the active generation.
+
+## Symlink tree profile
+
+Typical profiles created by [`nix-env`](@docroot@/command-ref/nix-env.md) (and indirectly by [`nix-channel`](@docroot@/command-ref/nix-channel.md) too) have additional structure.
+The contents of each generation is not just can arbitrary store object, but a symlink tree merging together various other store objects.
+
+Additionally, the store object contains a manifest file, providing some info on what those merged store objects are and where they came from:
+- [`manifest.nix`](@docroot@/command-ref/files/manifest.nix.md) used by [`nix-env`](@docroot@/command-ref/nix-env.md).
+- [`manifest.json`](@docroot@/command-ref/files/manifest.json.md) used by [`nix profile`](@docroot@/command-ref/new-cli/nix3-profile.md) (experimental).
+
+### Filesystem layout
```

Comment:
I used "typically" because, `nix-env --set` is the low level command to directly create a generation with arbitrary contents.

Perhaps it would make more sense to talk about the manifest and then the symlinking. The manifest describe the "input" to the symlinking algorithm.

Reply:
> I used "typically" because, `nix-env --set` is the low level command to directly create a generation with arbitrary contents.

Then let's be more concrete and say which subcommand does it and what's the difference. "Typically" is just too vague to say anything useful.